CONTRIBUTING.md

```diff
@@ -129,6 +129,38 @@ Run `rush build -t <packagename>` to build a single project, and all local proje
+
+By default, these npm scripts run recorded tests. We use a custom recording library called `test-utils-recorder`. You can learn more about our recorder at the [test-utils-recorder's readme](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md).
+
+#### Recorded tests
+
+By default, the npm scripts `test`, `test:node` and `test:browser` run recorded tests in playback mode i.e there is no request made to the service. For http requests made in each test case, there is a recorded response that is used which reproduces the service behavior.
+
+Keep in mind that if you modify any code that is being tested, or any test code, that interacts with the number of network calls that will be made during testing, or the shape of those network calls, the recordings will need to be re-generated.
+
+You can read more in the section: [Regenerating recordings](#regenerating-recordings).
+
+#### Live tests
+
+By default, the npm scripts `test`, `test:node` or `test:browser` run recorded tests. To run live tests, you need the right Azure resources, set appropriate environment variables to point to these resources and set the environment variable `TEST_MODE` to "live". The readme file in the test folder of each package lists the Azure resources and the environment variables needed.
+
+You can create the Azure resources manually on your own or automate this process by using the script called `New-TestResources.ps1`. Follow the steps in [`Example 1 of New-TestResources.ps1`](ttps://github.com/Azure/azure-sdk-for-js/blob/master/CONTRIBUTING.md#building). Once the project is correctly built, you will be able to run the tests by following the [testing instructions](https://github.com/Azure/azure-sdk-for-js/blob/master/eng/common/TestResources/New-TestResources.ps1.md#example-1) to set up a service principal and deploy live test resources. To see what resources will be deployed for a live service, check the `test-resources.json` ARM template files in the folder of the service you wish to deploy for testing, for example `sdk\keyvault\test-resources.json`.
+
+Live tests won't need updated recordings, and they also won't change previous recordings unless specified (see [Regenerating recordings](#regenerating-recordings)).
+
+Keep in mind that you can set the environment variables as how they're set by default through your system preferences, or through the command line, or by setting them each one in a different line in a `.env` file in the parent folder of the project you'll be testing (like in the folder `keyvault` if you're testing the project at `sdk/keyvault/keyvault-keys`). You can read more about how `dotenv` works in [their README](https://github.com/motdotla/dotenv#readme).
```

Comment:
Please see https://github.com/Azure/azure-sdk-for-js/pull/8398#discussion_r412646834

Reply:
I'll remove this paragraph.